src/Tools/ExternalAccess/FSharp/Classification/FSharpClassificationService.cs

```diff
@@ -5,62 +5,42 @@
 using System.Threading;
 using System.Threading.Tasks;
 using Microsoft.CodeAnalysis.Classification;
+using Microsoft.CodeAnalysis.ExternalAccess.FSharp.Classification;
 using Microsoft.CodeAnalysis.Host.Mef;
 using Microsoft.CodeAnalysis.Text;
-using Microsoft.CodeAnalysis.Editor;
-using Microsoft.CodeAnalysis.Host;
 
-namespace Microsoft.CodeAnalysis.ExternalAccess.Classification
+namespace Microsoft.CodeAnalysis.ExternalAccess.FSharp.Internal.Classification
 {
-    [ExportLanguageServiceFactory(typeof(IClassificationService), LanguageNames.FSharp), Shared]
-    internal class FSharpClassificationServiceFactory : ILanguageServiceFactory
+    [Shared]
+    [ExportLanguageService(typeof(IClassificationService), LanguageNames.FSharp)]
+    internal class FSharpClassificationService : IClassificationService
     {
-#pragma warning disable CS0618 // Type or member is obsolete
-        public ILanguageService CreateLanguageService(HostLanguageServices languageServices)
+        private readonly IFSharpClassificationService _service;
+
+        [ImportingConstructor]
```

Comment:
💡 Consider marking all MEF exports as obsolete to avoid having them constructed in unexpected ways:

```csharp
[Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
```

Reply:
Ok